tests/utilities/fault_tolerant.py

```diff
@@ -0,0 +1,143 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Here are the steps to use this file to ensure your cluster is properly setup for Fault Tolerant Training.
+
+1. Launch this script with `python tests/utilities/fault_tolerant.py`
+2. Once ``READY TO BE KILLED WITH SIGTERM SIGNAL`` is detected within
+   the Pod Logs, send a SIGTERM to this process. The process is waiting for it.
+3. You should detect `.pl_auto_save.ckpt exists` within the Pod Logs.
+4. Resume your instance with the same volumes and relaunch the same script.
+5. The script should contain `[-1.0939, -0.4306]` within its Pod Logs.
+
+The mode weights with the failure matches the weight without.
+The training has been properly resumed and is fully reproduced.
+
+Command: `python -m tests.utilities.fault_tolerant`
+"""
+
+# Note, this file is used to ensure Fault Tolerant is working as expected
+import os
+import random as python_random
+from contextlib import suppress
+from time import sleep
+
+import numpy as np
+import torch
+from torch.utils.data import DataLoader, Dataset
+
+from pytorch_lightning import _logger as log
+from pytorch_lightning import LightningModule, seed_everything, Trainer
+
+
+class Model(LightningModule):
+    def __init__(self, fail_on_step: int = -1):
+        super().__init__()
+        self.layer = torch.nn.Linear(1, 2)
+        self.seen_batches = []
+        self.fail_on_step = fail_on_step
+
+    def training_step(self, batch, batch_idx):
+        if self.global_step == self.fail_on_step:
+            log.info("READY TO BE KILLED WITH SIGTERM SIGNAL.")
+            while not self.trainer._terminate_gracefully:
+                sleep(0.00001)
```

Comment:
```suggestion
                sleep(0.1)
```

This seems like it'd burn many CPUs cycles for no good reason